CHANGES.rst

```diff
@@ -171,6 +183,13 @@ astropy.table
   were added. See the ``Masking change in astropy 4.0`` section within
   `<https://docs.astropy.org/en/v4.0/table/masking.html>`_ for details. [#8789]
 
+- Changed implementation of ``Table.add_column()`` and ``Table.add_columns()``
+  methods.  Now it is possible add any object(s) which can be converted or broadcasted
+  to a valid column for the table.  ``Table.__setitem__`` now just calls
+  ``add_column``.  One change is that a value with ``shape = (1,)`` is no longer
+  broadcasted to the column shape. E.g. ``t['a'] = [1]`` used to be the same as
```

Comment:
Do we really want this? It breaks analogy with an structured array, and it seems somewhat inconsistent with a scalar working. That said, the table is really different in that the first axis is treated as the length. But I don't like special-casing the 1-d case...

Reply:
OK I can revert this.  Just always seemed to be weird to me, but I don't feel that strongly, and fewer API changes is good.